src/providers/pkcs11/generate_random.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::utils::to_response_status;
+use super::Provider;
+use parsec_interface::operations::psa_generate_random;
+use parsec_interface::requests::{ResponseStatus, Result};
+use std::convert::TryFrom;
+
+impl Provider {
+    pub(super) fn psa_generate_random_internal(
+        &self,
+        op: psa_generate_random::Operation,
+    ) -> Result<psa_generate_random::Result> {
+        let length = u32::try_from(op.size).or_else(|_| {
+            let error = ResponseStatus::ResponseTooLarge;
+            format_error!("Generate random status", error);
```

Comment:
Given that we can't be very specific with the error code above, this should probably just be something like
```
error!("Requested size is too large");
```

Reply:
Will update the print log.